### PR TITLE
Added saving of features

### PR DESCRIPTION
**Pull request by deactivated-acc**
New button in imagedisplay activity prompts for filename and saves
keypoints and descriptors as json

As described in issue #17

### CODE REVIEW

**Comment by jrauhamaa** [comment]
Since we are only saving data files here, do you think it makes sense to ask the user for file names?

File: FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java [hunk 1]
```diff
@@ -17,6 +48,201 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            fw.flush();
+            fw.close();
+        } catch (Exception e){
+            e.printStackTrace();
+        }
+
+        try{
+            FileWriter fw = null;
+
+            File keypointsFile = new File(fileDir, fileName+"_keypoints");
```
Comment (jrauhamaa) [inline FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java]: Is this code duplicated on purpose?

**Review by jorenham — APPROVED** [review]

**Review by jrauhamaa — CHANGES_REQUESTED** [review]
- remove duplicated code
- would be nice if the code would generate automatic filename (e.g. an uuid) instead of asking the user given that the user won't be able to access the files. 
- would be nice if the user got notified of successful save

**Comment by deactivated-acc** [comment]
-Added loading files for the matching part
-Removed the duplicate code in saveFeatures
-Added toast messages for save success and failure

@jrauhamaa automatic filename generation isn't really useful right now, but can be easily added when it is necessary. With the known filename, you can easily see which one to match with, which you can't with the automatically generated one.

**Comment by jorenham** [comment]
You got some conflict

**Pull request merged**